agent/worker.go

```diff
@@ -344,6 +357,49 @@ func reconcileSecrets(ctx context.Context, w *worker, assignments []*api.Assignm
 	return nil
 }
 
+func reconcileConfigs(ctx context.Context, w *worker, assignments []*api.AssignmentChange, fullSnapshot bool) error {
```

Comment:
Would `reconcileSet` just call both functions or do you have a more significant refactor in mind?